src/qfusion.cpp

```diff
@@ -0,0 +1,541 @@
+//////////////////////////////////////////////////////////////////////////////////////
+//
+// (C) Daniel Strano and the Qrack contributors 2017, 2018. All rights reserved.
+//
+// QUnitMulti is a multiprocessor variant of QUnit.
+// QUnit maintains explicit separability of qubits as an optimization on a QEngine.
+// See https://arxiv.org/abs/1710.05867
+// (The makers of Qrack have no affiliation with the authors of that paper.)
+//
+// Licensed under the GNU Lesser General Public License VMIN_FUSION_BITS.
+// See LICENSE.md in the project root or https://www.gnu.org/licenses/lgpl-3.0.en.html
+// for details.
+
+#include <ctime>
+#include <future>
+#include <initializer_list>
+#include <map>
+
+#include "qfactory.hpp"
+#include "qfusion.hpp"
+
+namespace Qrack {
+
+QFusion::QFusion(
+    QInterfaceEngine eng, bitLenInt qBitCount, bitCapInt initState, std::shared_ptr<std::default_random_engine> rgp)
+    : QInterface(qBitCount)
+    , engineType(eng)
+    , bitBuffers(qBitCount)
+{
+    if (rgp == nullptr) {
+        /* Used to control the random seed for all allocated interfaces. */
+        rand_generator = std::make_shared<std::default_random_engine>();
+        rand_generator->seed(std::time(0));
+    } else {
+        rand_generator = rgp;
+    }
+
+    qReg = CreateQuantumInterface(engineType, qBitCount, initState, rand_generator);
+}
+
+void QFusion::ApplySingleBit(const complex* mtrx, bool doCalcNorm, bitLenInt qubitIndex)
```

Comment:
Add some comments throughout this code talking about what's going on, and (more importantly) why.

Reply:
See my next commit. It's a simple concept, but I've hopefully documented it fully in the commit to follow.